src/error.rs

```diff
@@ -49,7 +49,72 @@ pub struct Pos {
     path: String,
 }
 
+
+/// This type represents the location that an error occured
+#[derive(Debug)]
+pub struct ErrorMarker {
```

Comment:
Could you brainstorm some alternative names for this? [`marker`](https://doc.rust-lang.org/std/marker/index.html) means something else in Rust.